packages/swapper/src/swappers/ChainflipSwapper/endpoints.ts

```diff
@@ -0,0 +1,347 @@
+import { fromAssetId, fromChainId, solAssetId } from '@shapeshiftoss/caip'
+import type { BuildSendApiTxInput, GetFeeDataInput } from '@shapeshiftoss/chain-adapters'
+import { FeeDataKey } from '@shapeshiftoss/chain-adapters'
+import type { BTCSignTx, SolanaSignTx } from '@shapeshiftoss/hdwallet-core'
+import type { EvmChainId, KnownChainIds } from '@shapeshiftoss/types'
+import { TxStatus } from '@shapeshiftoss/unchained-client'
+import type { AxiosError } from 'axios'
+import type { InterpolationOptions } from 'node-polyglot'
+
+import type {
+  EvmTransactionRequest,
+  GetUnsignedEvmTransactionArgs,
+  GetUnsignedSolanaTransactionArgs,
+  GetUnsignedUtxoTransactionArgs,
+  SwapperApi,
+  UtxoFeeData,
+} from '../../types'
+import { isExecutableTradeQuote, isExecutableTradeStep, isToken } from '../../utils'
+import { CHAINFLIP_BAAS_COMMISSION, chainIdToChainflipNetwork } from './constants'
+import type { ChainflipBaasSwapDepositAddress } from './models/ChainflipBaasSwapDepositAddress'
+import { getTradeQuote, getTradeRate } from './swapperApi/getTradeQuote'
+import type { ChainFlipStatus } from './types'
+import { chainflipService } from './utils/chainflipService'
+import { getLatestChainflipStatusMessage } from './utils/getLatestChainflipStatusMessage'
+
+// Persists the ID so we can look it up later when checking the status
+const tradeQuoteMetadata: Map<string, ChainflipBaasSwapDepositAddress> = new Map()
+
+export const chainflipApi: SwapperApi = {
+  getTradeQuote,
+  getTradeRate,
+  getUnsignedEvmTransaction: async ({
+    chainId,
+    from,
+    tradeQuote,
+    assertGetEvmChainAdapter,
+    config,
+    supportsEIP1559,
+  }: GetUnsignedEvmTransactionArgs): Promise<EvmTransactionRequest> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+
+    const brokerUrl = config.REACT_APP_CHAINFLIP_API_URL
+    const apiKey = config.REACT_APP_CHAINFLIP_API_KEY
+
+    const step = tradeQuote.steps[0]
+    const sellChainflipChainKey = `${step.sellAsset.symbol.toLowerCase()}.${
```

Comment:
No idea if you also want this to be in an `assetIdToChainflipAsset` map? It works for the current tokens with just the symbol for now :)

Reply:
Yep totally fine given the low amount of supported assets IMO

Reply:
akschually: d097f8c50149b80579acff700bc7f8e2afbb2a41